app/scripts/metamask-controller.js

```diff
@@ -2417,6 +2441,17 @@ export default class MetamaskController extends EventEmitter {
 
         for (const [origin, accounts] of changedAccounts.entries()) {
           this._notifyAccountsChange(origin, accounts);
+          // When permissions are added for a given origin we need to
+          // set the network client id for that origin to the current
+          // globally selected network client id.
+          if (accounts.length > 0) {
+            this.selectedNetworkController.setNetworkClientIdForDomain(
+              origin,
+              this.networkController.state.selectedNetworkClientId,
+            );
```

Comment:
this is only necessary because we access domain state directly, right? because `getNetworkClientIdForDomain()` would already return the selectedNetworkClientId as a fallback

Reply:
So this is actually the point at which we want to add the newly connected dapp to the `domains` state in `SelectedNetworkController`. 

However, this is another candidate for something that can/should be pulled directly into the `SelectedNetworkController`

Reply:
Sounds good. the less MetaMaskController has to be concerned about, the better